packages/jss-plugin-rule-value-function/src/index.js

```diff
@@ -16,6 +17,34 @@ const fnRuleNs = `fnStyle${++now}`
 
 type StyleRuleWithRuleFunction = StyleRule & {[key: string]: Function}
 
+function getFunctionStyles(styles: Object): Object | null {
+  let to = null
+
+  for (const key in styles) {
+    const value = styles[key]
+    const type = typeof value
+
+    if (isObservable(value)) {
```

Comment:
Instead of having the knowledge about observables here, we could detect non-plain objects and ignore them all:
Here is the lodash implementation https://github.com/lodash/lodash/blob/master/isPlainObject.js

I think it would be already enough:

```js
const plainObjectConstrurctor = {}.constructor

if (value != null && value.constructor !== plainObjectConstrurctor) {
  continue;
}
```

Reply:
Yeah, I wasn't to happy about that either.

Reply:
Maybe we could use that also in the core `cloneStyle`?

Reply:
true

Reply:
wait, if we filter all non-plain objects in the core, we won't let Observables pass through to the plugin.